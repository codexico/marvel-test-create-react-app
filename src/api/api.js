import axios from 'axios';

import { handleError } from './api-utils';

const apikey = process.env.REACT_APP_API_KEY;
const path = 'https://gateway.marvel.com:443/v1/public/characters';

export const apiHeroesList = () => {
  const url = `${path}?apikey=${apikey}`;

  return axios.get(url).catch((err) => handleError(err));
};

export const apiHeroByID = (heroID) => {
  const url = `${path}/${heroID}?apikey=${apikey}`;

  return axios.get(url).catch((err) => handleError(err));
};
