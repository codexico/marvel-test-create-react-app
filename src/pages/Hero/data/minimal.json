{
    "code": 200,
    "status": "Ok",
    "copyright": "© 2020 MARVEL",
    "attributionText": "Data provided by Marvel. © 2020 MARVEL",
    "attributionHTML": "<a href=\"http://marvel.com\">Data provided by Marvel. © 2020 MARVEL</a>",
    "etag": "75b6c8824c323dc143d1104b505f3ce854a0927c",
    "data": {
      "offset": 0,
      "limit": 20,
      "total": 1,
      "count": 1,
      "results": [
        {
          "id": 1010356,
          "name": "Black Queen",
          "description": "",
          "modified": "1969-12-31T19:00:00-0500",
          "thumbnail": {
            "path": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available",
            "extension": "jpg"
          },
          "resourceURI": "http://gateway.marvel.com/v1/public/characters/1010356",
          "comics": {
            "available": 2,
            "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010356/comics",
            "items": [
              {
                "resourceURI": "http://gateway.marvel.com/v1/public/comics/7750",
                "name": "Captain America (1968) #369"
              },
              {
                "resourceURI": "http://gateway.marvel.com/v1/public/comics/18136",
                "name": "X-Men Unlimited (1993) #33"
              }
            ],
            "returned": 2
          },
          "series": {
            "available": 2,
            "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010356/series",
            "items": [
              {
                "resourceURI": "http://gateway.marvel.com/v1/public/series/1996",
                "name": "Captain America (1968 - 1996)"
              },
              {
                "resourceURI": "http://gateway.marvel.com/v1/public/series/3637",
                "name": "X-Men Unlimited (1993 - 2003)"
              }
            ],
            "returned": 2
          },
          "stories": {
            "available": 2,
            "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010356/stories",
            "items": [
              {
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/18018",
                "name": "CAPTAIN AMERICA (1968) #369",
                "type": "cover"
              },
              {
                "resourceURI": "http://gateway.marvel.com/v1/public/stories/67737",
                "name": "X-Men Unlimited 33 cover",
                "type": "cover"
              }
            ],
            "returned": 2
          },
          "events": {
            "available": 0,
            "collectionURI": "http://gateway.marvel.com/v1/public/characters/1010356/events",
            "items": [],
            "returned": 0
          },
          "urls": [
            {
              "type": "detail",
              "url": "http://marvel.com/characters/278/black_queen?utm_campaign=apiRef&utm_source=e34a7870997300d416ccd63679bebfe0"
            },
            {
              "type": "comiclink",
              "url": "http://marvel.com/comics/characters/1010356/black_queen?utm_campaign=apiRef&utm_source=e34a7870997300d416ccd63679bebfe0"
            }
          ]
        }
      ]
    }
  }
