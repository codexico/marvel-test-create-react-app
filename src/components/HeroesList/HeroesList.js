import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import { apiHeroesList, apiHeroesListOrderByName } from '../../api/api';

import HeroCard from '../HeroCard/HeroCard';

import './HeroesList.css';

function HeroesList({ handleClick }) {
  const [characters, setCharacters] = useState();
  const [ascDesc, setAscDesc] = useState(false);

  useEffect(() => {
    apiHeroesList().then(({ data }) => setCharacters(data));
  }, []);

  function orderByName() {
    apiHeroesListOrderByName(ascDesc).then(({ data }) => setCharacters(data));
    setAscDesc(!ascDesc);
  }

  return (
    <div className="HeroesList">
      <div className="HeroesList-options">
        <button onClick={orderByName}>
          Ordenar por nome - A/Z
      </button>
      </div>
      <div className="HeroesList-cards">
        {characters?.data?.results?.map((character) => (
          <HeroCard
            key={character.id}
            heroID={character.id}
            name={character.name}
            thumbnail={character.thumbnail}
            handleClick={handleClick}
          />
        ))}
      </div>
    </div>
  );
}

HeroesList.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default HeroesList;
