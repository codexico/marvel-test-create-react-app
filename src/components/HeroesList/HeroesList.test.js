import React from 'react';
import {
  cleanup,
  render,
  waitForElement,
  fireEvent,
} from '@testing-library/react';

import data from './data/characters_limit20.json';
import dataOrder from './data/characters_order-name.json';

import { apiHeroesList, apiHeroesListOrderByName } from '../../api/api';

jest.mock('../../api/api');

import HeroesList from './HeroesList';

afterEach(() => {
  apiHeroesList.mockClear();
  cleanup();
});

test('Component renders correctly', async () => {
  apiHeroesList.mockResolvedValue({ data: data });

  const { asFragment, getByText } = render(
    <HeroesList handleClick={() => {}} />
  );

  expect(apiHeroesList).toHaveBeenCalledTimes(1);

  const hero = await waitForElement(() => getByText('Hulk'));

  expect(hero).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('Component reorders', async () => {
  apiHeroesList.mockResolvedValue({ data: data });

  const { asFragment, getByText } = render(
    <HeroesList handleClick={() => {}} />
  );

  expect(apiHeroesList).toHaveBeenCalledTimes(1);
  expect(asFragment()).toMatchSnapshot();

  apiHeroesListOrderByName.mockResolvedValueOnce({ data: data });
  const orderButton = await waitForElement(() =>
    getByText('Ordenar por nome - A/Z')
  );
  fireEvent.click(orderButton);
  const hero = await waitForElement(() => getByText('Hulk'));

  expect(hero).toBeInTheDocument();
  expect(apiHeroesListOrderByName).toHaveBeenCalledTimes(1);
  expect(apiHeroesListOrderByName).toHaveBeenCalledWith(false);
  expect(asFragment()).toMatchSnapshot();

  apiHeroesListOrderByName.mockResolvedValueOnce({ data: dataOrder });
  const orderButtonReverse = await waitForElement(() =>
    getByText('Ordenar por nome - A/Z')
  );
  fireEvent.click(orderButtonReverse);
  const heroReverse = await waitForElement(() => getByText('Zzzax'));

  expect(heroReverse).toBeInTheDocument();
  expect(apiHeroesListOrderByName).toHaveBeenCalledTimes(2);
  expect(apiHeroesListOrderByName).toHaveBeenCalledWith(true);
  expect(asFragment()).toMatchSnapshot();
});
