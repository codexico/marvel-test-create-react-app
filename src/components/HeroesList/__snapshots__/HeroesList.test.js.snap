// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="HeroesList"
  >
    <div
      class="HeroesList-options"
    >
      <button>
        Ordenar por nome - A/Z
      </button>
    </div>
    <div
      class="HeroesList-cards"
    >
      <div
        class="HeroCard"
      >
        <img
          alt="Hulk"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/a0/538615ca33ab0.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Hulk
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Mercury"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Mercury
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Nicolaos"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Nicolaos
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="The Executioner"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/e/d0/4ce5a2ab860be.jpg"
        />
        <p
          class="HeroCard-name"
        >
          The Executioner
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Champions"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/8/b0/4c00376da82bb.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Champions
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Jack Murdock"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Jack Murdock
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Janus, the Nega-Man"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Janus, the Nega-Man
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Mordo"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Mordo
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Mysterio (Francis Klum)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Mysterio (Francis Klum)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Sabretooth (House of M)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/80/4c0033dd26d33.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Sabretooth (House of M)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Lady Mastermind"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/d0/4c00326f63d4c.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Lady Mastermind
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Living Mummy"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Living Mummy
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Living Tribunal"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/2/a0/4c0031062f91f.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Living Tribunal
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Scorpion (Ultimate)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Scorpion (Ultimate)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Omega the Unknown"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/f/10/4ce5a4e4cd15d.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Omega the Unknown
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Infant Terrible"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/2/f0/4c002f7854e02.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Infant Terrible
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Crimson King"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Crimson King
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Captain Cross"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Captain Cross
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="War Machine (Ultimate)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/9/a0/4ce5a5647d38b.jpg"
        />
        <p
          class="HeroCard-name"
        >
          War Machine (Ultimate)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Patch"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Patch
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component reorders 1`] = `
<DocumentFragment>
  <div
    class="HeroesList"
  >
    <div
      class="HeroesList-options"
    >
      <button>
        Ordenar por nome - A/Z
      </button>
    </div>
    <div
      class="HeroesList-cards"
    />
  </div>
</DocumentFragment>
`;

exports[`Component reorders 2`] = `
<DocumentFragment>
  <div
    class="HeroesList"
  >
    <div
      class="HeroesList-options"
    >
      <button>
        Ordenar por nome - A/Z
      </button>
    </div>
    <div
      class="HeroesList-cards"
    >
      <div
        class="HeroCard"
      >
        <img
          alt="Hulk"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/a0/538615ca33ab0.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Hulk
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Mercury"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Mercury
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Nicolaos"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Nicolaos
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="The Executioner"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/e/d0/4ce5a2ab860be.jpg"
        />
        <p
          class="HeroCard-name"
        >
          The Executioner
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Champions"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/8/b0/4c00376da82bb.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Champions
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Jack Murdock"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Jack Murdock
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Janus, the Nega-Man"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Janus, the Nega-Man
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Mordo"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Mordo
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Mysterio (Francis Klum)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Mysterio (Francis Klum)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Sabretooth (House of M)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/80/4c0033dd26d33.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Sabretooth (House of M)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Lady Mastermind"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/d0/4c00326f63d4c.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Lady Mastermind
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Living Mummy"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Living Mummy
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Living Tribunal"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/2/a0/4c0031062f91f.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Living Tribunal
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Scorpion (Ultimate)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Scorpion (Ultimate)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Omega the Unknown"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/f/10/4ce5a4e4cd15d.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Omega the Unknown
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Infant Terrible"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/2/f0/4c002f7854e02.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Infant Terrible
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Crimson King"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Crimson King
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Captain Cross"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Captain Cross
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="War Machine (Ultimate)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/9/a0/4ce5a5647d38b.jpg"
        />
        <p
          class="HeroCard-name"
        >
          War Machine (Ultimate)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Patch"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Patch
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Component reorders 3`] = `
<DocumentFragment>
  <div
    class="HeroesList"
  >
    <div
      class="HeroesList-options"
    >
      <button>
        Ordenar por nome - A/Z
      </button>
    </div>
    <div
      class="HeroesList-cards"
    >
      <div
        class="HeroCard"
      >
        <img
          alt="Zzzax"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/c/d0/4ced5ab9078c9.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zzzax
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zuras"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zuras
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zombie (Simon Garth)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/6/10/4c003937c9ba4.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zombie (Simon Garth)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zodiak"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zodiak
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zemo"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zemo
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zeigeist"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zeigeist
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zarek"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/90/4c002f38d0e05.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zarek
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zarda"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zarda
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zaran"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zaran
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Zaladane"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Zaladane
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Young X-Men"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Young X-Men
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Young Avengers"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/c0/50fec602eed20.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Young Avengers
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Yellowjacket (Rita DeMara)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/9/f0/5398a8a4b8ca9.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Yellowjacket (Rita DeMara)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Yellow Claw"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Yellow Claw
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Xorn (Kuan-Yin Xorn)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/d/60/4c00394059727.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Xorn (Kuan-Yin Xorn)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="Xavin"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/3/a0/4ce5a49743462.jpg"
        />
        <p
          class="HeroCard-name"
        >
          Xavin
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="X.S.E."
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
        />
        <p
          class="HeroCard-name"
        >
          X.S.E.
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="X-Statix"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/5/40/4c003724ba21f.jpg"
        />
        <p
          class="HeroCard-name"
        >
          X-Statix
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="X-Ray (James Darnell)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/f/40/4c0035a250615.jpg"
        />
        <p
          class="HeroCard-name"
        >
          X-Ray (James Darnell)
        </p>
      </div>
      <div
        class="HeroCard"
      >
        <img
          alt="X-Men (Ultimate)"
          class="HeroCard-img"
          src="http://i.annihil.us/u/prod/marvel/i/mg/8/a0/4feb34f2017a0.jpg"
        />
        <p
          class="HeroCard-name"
        >
          X-Men (Ultimate)
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
